### DISCUSSION TITLE
Is it expected to fully disable conceal when entering Visual mode?

### DISCUSSION THREAD

**Discussion by rodhash** [MeanderingProgrammer/render-markdown.nvim#251] [General]
It's not disabling conceal only in the current line but in the whole file.

Is it expected?

Is it possible to disable conceal only in the current / selected lines?

<img src="https://github.com/user-attachments/assets/51bce318-2016-4546-a205-0ad2351f0035" width="258">

Update:

It seems adding "v" to `render_modes` almost do the trick, the one I need to add there is actually `V-LINE` and `V-BLOCK` but not sure it supports this value.

Readme says:
```
    -- Vim modes that will show a rendered view of the markdown file, :h mode()
    -- All other modes will be unaffected by this plugin
    render_modes = { 'n', 'c', 't' },
```
Interesting that `v` also seems to work in there.
Thanks

**Comment by MeanderingProgrammer** [comment]
All mode values are supported, you just need to use the correct values from `:h mode()`. The block one `CTRL-V` is the only special case, if you want to add all 3 visual modes you would do:

```lua
require('render-markdown').setup({
    render_modes = { 'n', 'c', 't', 'v', 'V', '\22'  },
})
```

- `v`: visual character
- `V`: visual line
- `\22`: visual block (need to use the ascii code for this one)

**Reply by rodhash** [reply]
> * `v`: visual character
> * `V`: visual line
> * `\22`: visual block (need to use the ascii code for this one)

Thanks, worked!

**Reply by rodhash** [reply]
The only thing I noticed is, right when I enter visual mode it stills disables conceal:

<img src="https://github.com/user-attachments/assets/793efb83-5a7a-4f43-8cd7-a8a63fbc8e7a" width="300">


And conceal stays disabled until I move the cursor to select other lines.

**Reply by MeanderingProgrammer** [reply]
Will need a minimal reproducible example. Could be another plugin, could be a configuration, could be the version you're running. Does not happen on my end.

**Reply by rodhash** [reply]
Yeah with a minimal this is not happening.. I need to find which plugin is causing this.

```lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "nvim-treesitter/nvim-treesitter",
  "MeanderingProgrammer/render-markdown.nvim"
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})



require'nvim-treesitter.configs'.setup {
  -- A list of parser names, or "all" (the listed parsers MUST always be installed)
  ensure_installed = { "markdown", "markdown_inline" },

  -- Install parsers synchronously (only applied to `ensure_installed`)
  sync_install = false,

  -- Automatically install missing parsers when entering buffer
  -- Recommendation: set to false if you don't have `tree-sitter` CLI installed locally
  auto_install = true,


  ---- If you need to change the installation directory of the parsers (see -> Advanced Setup)
  -- parser_install_dir = "/some/path/to/store/parsers", -- Remember to run vim.opt.runtimepath:append("/some/path/to/store/parsers")!

  highlight = {
    enable = true,
  },
}

require('render-markdown').setup({
    render_modes = { 'i', 'n', 'c', 't', 'v', 'V', '\22'  },
})
```

BTW when we hit `>` in order to indent a paragraph Nvim enters in `O-PENDING` mode, is it supported in `render_modes` too?

**Reply by MeanderingProgrammer** [reply]
Yes, anything you see in `:h mode()` is a valid value for `render_modes`. Operator-pending is the second mode listed in the help doc.
